clap_builder/src/builder/value_hint.rs

```diff
@@ -65,6 +65,35 @@ pub enum ValueHint {
     Url,
     /// Email address.
     EmailAddress,
+
+    /// Custom value hint. This is used to provide a hint that is not covered by the predefined
+    Custom,
+}
+
+/// This trait is used to provide users a way to add custom value hint to the argument.
+/// This is useful when predefined value hints are not enough.
+pub trait CustomValueHint: core::fmt::Debug + Send + Sync + CustomValueHintClone {
+    /// This method should return a list of custom value hints.
+    fn custom_hint(&self) -> Option<Vec<OsString>>;
```

Comment:
Whats the role of `Option` with this?

Reply:
I think user-defined code may not generate completion results. For example, if external resources cannot be accessed, it could return `None` instead of an empty `Vec`.

Reply:
What do we expect to do differently in that scenario?

btw the result of this conversation would be good to capture in the doc comment.

Reply:
Thanks for your review! `None` and `Some(vec![])` are in the same state, which may confuse users.